### PR TITLE
wdio5: fix switch frame on MS Edge

### PR DESCRIPTION
**Pull request by kyryloonufriiev**
This is a very small PR related to this card [Trello] https://trello.com/c/SLUduLu8/68-can-take-baseline-screenshot-but-checkpoint-screenshots-not-showing-up.

As far as I understand the customer faces an issue when the driver is trying to switch frame it throws an error `ERROR webdriver: Request failed due to no such frame: The specified frame was not found in the current frame or window.`.
I see that an earlier version was a bug on `EdgeDriver` but during my investigation, I realize that the problem on the wdio5 itself. When we call `browser.switchToFrame(element)`, as an element we pass result of `browser.$(selector)` (according to our implementation). So for the MS Edge browser it returns
```
Element {
  sessionId: 'a2221fa8a86ec5320835cf3a8b27fd211795d64f',
  elementId: '6dedc078-5b3e-49f3-a879-387bd5a7d537',
  'element-6066-11e4-a52e-4f735466cecf': '6dedc078-5b3e-49f3-a879-387bd5a7d537',
  selector: '/html[1]/body[1]/iframe[1]',
  ...
}
```
In the same time for Chrome it will be
```
Element {
  sessionId: 'a2221fa8a86ec5320835cf3a8b27fd211795d64f',
  elementId: '6dedc078-5b3e-49f3-a879-387bd5a7d537',
  ELEMENT: '6dedc078-5b3e-49f3-a879-387bd5a7d537',
  'element-6066-11e4-a52e-4f735466cecf': '6dedc078-5b3e-49f3-a879-387bd5a7d537',
  selector: '/html[1]/body[1]/iframe[1]',
  ...
}
```
And only thing is matter for `browser.switchToFrame(element)`, is an element object should have `ELMENT` field with valid id.

### CODE REVIEW

**Comment by tourdedave** [comment]
Is there a test reference for this?

**Comment by kyryloonufriiev** [comment]
I'm not sure how I can add test for Edge. I used browserstack credentials to test it. But I think it shouldn't be shared through GitHub. Or do you want me to moch Edge responses some how?

**Comment by tourdedave** [comment]
You can add the test with credentials passed as an environment variable
(e.g., BROWSERSTACK_USERNAME, BROWSERSTACK_ACCESS_KEY). I would add it to
test/coverage/custom/.

בתאריך יום ה׳, 26 במרץ 2020 ב-12:59 מאת kyryloonufriiev <
notifications@github.com>:

**Comment by amitzur** [comment]
> We'll want to confirm that the coverage tests that cover frames pass (not sure which of them are active as part of a release right now).

@tourdedave do you mean we should see if we can enable more tests that are currently enabled?

**Comment by tourdedave** [comment]
> > We'll want to confirm that the coverage tests that cover frames pass (not sure which of them are active as part of a release right now).
> 
> @tourdedave do you mean we should see if we can enable more tests that are currently enabled?

@amitzur Yes and no. My point was that we'll want to sure that standard flows for frames work with this change. I know we have tests for this in the coverage tests. Not sure if they're enabled since what is enabled has been in flux over the last week. Worth enabling them if they're disabled.

File: packages/eyes-webdriverio-5/src/wrappers/EyesTargetLocator.js [hunk 1]
```diff
@@ -92,27 +86,16 @@ class EyesTargetLocator extends TargetLocator {
...
-    let frameElement = arg1
     this._logger.verbose('EyesTargetLocator.frame(element)')
     this._logger.verbose('Making preparations...')
     await this.willSwitchToFrame(frameElement)
     this._logger.verbose('Done! Switching to frame...')
-    if (frameElement instanceof EyesWebElement) {
-      frameElement = frameElement.getWebElement()
-    }
-    await this._targetLocator.frame(frameElement.element)
+    await this._targetLocator.frame(frameElement.jsonElement)
```
[the remark this answers is not in the document]
Comment (amitzur) [reply packages/eyes-webdriverio-5/src/wrappers/EyesTargetLocator.js]: Just because it existed before, I think I'd want to support here also a `WebElement` instance.
Also, in wdio4 I added a function called `isWDIOElement`. It should probably be revisited, since it's doing this:
```
function isWDIOElement(object) {
  return object && object.ELEMENT
}
```

File: packages/eyes-webdriverio-5/src/wrappers/WebElement.js [hunk 2]
```diff
@@ -4,6 +4,8 @@ const {ArgumentGuard, TypeUtils} = require('@applitools/eyes-sdk-core')
 const command = require('../services/selenium/Command')
 const {By} = require('../By')
 
+const WEB_ELEMENT_IDENTIFIER = 'element-6066-11e4-a52e-4f735466cecf'
```
[the remark this answers is not in the document]
Comment (amitzur) [reply packages/eyes-webdriverio-5/src/wrappers/WebElement.js:7]: oh wow, good to know!
